src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -41,51 +41,106 @@ const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
-  const initialMetadata: BootableVolumeMetadata = useMemo(() => {
+
+  const instanceTypesAndSizesObject = useMemo(
```

Comment:
How would you feel about changing the name to `instanceTypesToSizesMap`? I just looked and it appears we're passing instanceTypesNames down through a lot of components, but I'm not sure it's entirely necessary. We can look at that outside of this PR, though.

Reply:
> I just looked and it appears we're passing instanceTypesNames down through a lot of components

I looked into that and there isn't any case where it wouldn't be useless to pass it down thru lots of components. We need it in many places. Also note that we are doing it to minimize the amount of calls to the server for this resource, so we call it once in the parent component and pass it down to the other components that need the same info.